crates/xilem_web/src/context.rs

```diff
@@ -41,12 +41,52 @@ fn remove_attribute(element: &web_sys::Element, name: &str) {
     }
 }
 
+fn set_class(element: &web_sys::Element, class_name: &str) {
+    #[cfg(debug_assertions)]
+    if class_name.is_empty() {
+        panic!("class names cannot be the empty string");
+    }
+    #[cfg(debug_assertions)]
+    if class_name.contains(' ') {
+        panic!("class names cannot contain the ascii space character");
+    }
+    element.class_list().add_1(class_name).unwrap_throw()
+}
+
+fn remove_class(element: &web_sys::Element, class_name: &str) {
+    #[cfg(debug_assertions)]
+    if class_name.is_empty() {
+        panic!("class names cannot be the empty string");
+    }
+    #[cfg(debug_assertions)]
+    if class_name.contains(' ') {
+        panic!("class names cannot contain the ascii space character");
+    }
+    element.class_list().remove_1(class_name).unwrap_throw()
+}
+
+fn set_style(element: &web_sys::Element, name: &str, value: &str) {
+    // styles will be ignored for non-html elements (e.g. SVG)
+    if let Some(el) = element.dyn_ref::<web_sys::HtmlElement>() {
+        el.style().set_property(name, value).unwrap_throw()
+    }
+}
+
+fn remove_style(element: &web_sys::Element, name: &str) {
+    // styles will be ignored for non-html elements (e.g. SVG)
+    if let Some(el) = element.dyn_ref::<web_sys::HtmlElement>() {
+        el.style().remove_property(name).unwrap_throw();
+    }
+}
+
 // Note: xilem has derive Clone here. Not sure.
 pub struct Cx {
     id_path: IdPath,
     document: Document,
     // TODO There's likely a cleaner more robust way to propagate the attributes to an element
     pub(crate) current_element_attributes: VecMap<CowStr, AttributeValue>,
+    pub(crate) current_element_classes: VecMap<CowStr, ()>,
+    pub(crate) current_element_styles: VecMap<CowStr, CowStr>,
```

Comment:
Not that it is particularly relevant to this PR, more a brain-dump/thoughts about it, I'd like to share.

I'm more and more thinking about getting the temporary state of elements out of the context again, likely via associated type bounds as described [here](https://xi.zulipchat.com/#narrow/stream/354396-xilem/topic/xilem_html.3A.20strong.20typing.20.28attributes.20and.20elements.29/near/387972736), since they're [stabilized](https://github.com/rust-lang/rust/pull/122055) now. This would allow (safe) implementations downstream (i.e. remove the "need" for the `Sealed` trait in the interfaces), better separation of concerns (especially if there's a lot of logic added for all kinds of dom elements), possibly it would also help stripping down to only relevant code (e.g. when style is not used, it could be stripped away in the resulting binary).
And generally make everything cleaner I think. I think it's probably worth the added compilation times. But this probably needs deeper investigation/experimentation (again).